src/elastio-snap.c

```diff
@@ -2867,12 +2866,14 @@ static int bio_make_read_clone(struct block_device *bdev, struct bio_set *bs, st
 	bio_sector(new_bio) = sect;
 	bio_idx(new_bio) = 0;
 
-#ifdef BIO_REDIRECT_TO_PART0
-	/**
-	 * starting from 5.16+ we change the device to the root partition
-	 * to make sure the 'dev->sd_sect_off' math remains the same
-	 */
-	new_bio->bi_bdev = new_bio->bi_bdev->bd_disk->part0;
+#if defined HAVE_BIO_REMAPPED
+	if (bio_flagged(orig_bio, BIO_REMAPPED))
+		bio_set_flag(new_bio, BIO_REMAPPED);
+#endif
+
+#if defined HAVE_BIO_THROTTLED
+	if (bio_flagged(orig_bio, BIO_THROTTLED))
+		bio_set_flag(new_bio, BIO_THROTTLED);
```

Comment:
It would be great to leave here some comment about influence of these flags to the attempt to access beyond end of device =)

Reply:
Done.